linc_convert/modalities/psoct/_utils.py

```diff
@@ -203,17 +203,24 @@ def generate_pyramid(
                 slice(i * max_load, min((i + 1) * max_load, n))
                 for i, n in zip(chunk_index, prev_shape)
             ]
+            fullshape = omz[str(level - 1)].shape
             dat = omz[str(level - 1)][tuple(slicer)]
 
             # Discard the last voxel along odd dimensions
-            crop = [0 if x == 1 else x % 2 for x in dat.shape[-ndim:]]
+            crop = [
```

Comment:
Could you please explain what is this for? Why do we need to use this full shape instead? Since this changes breaks another test from another modality.

Reply:
We divide the previous resolution by 2. Thinking in 1D, to do this we reshape a dimension [N] into a dimension [N//2, 2] (essentialy you get a stack of the odd and even voxels) and then average across the new small axis.

However, we can only do this if the original dimension is even. So if it's odd I crop the last voxel (by doing something like array[:-1]).

That said, I think that the current code assumes that the data is exactly 3D (no channel dimension). I might have fixed it in the other PR (to be merged).

What error do you get, and on what kind of data?

Reply:
I see. The error from another modality is basically dimension not matching which makes sense to me now. Could you please point me to the other PR if it is not the one we just merged?

Reply:
It was the one you just merged